app/frontend/helpers/catalyst.ts

```diff
@@ -0,0 +1,56 @@
+import * as pbkdf2 from 'pbkdf2'
+import * as chacha from 'chacha'
+
+// TODO: This is what yoroi uses - rewrite with packages we use
+
+const PBKDF_ITERATIONS = 12983
+const SALT_SIZE = 16
+const KEY_SIZE = 32
+const DIGEST = 'sha512'
+const NONCE_SIZE = 12
+const PROTO_VERSION = Buffer.from('01', 'hex')
+
+/*
+	----------------------------------------------------------
+	| 0x01 | SALT(16) | NONCE(12) | Encrypted Data | Tag(16) |
+	----------------------------------------------------------
+*/
+
+const promisifyPbkdf2: (Uint8Array, Buffer) => Promise<Buffer> = (password, salt) => {
+  return new Promise((resolve, reject) => {
+    // eslint-disable-next-line consistent-return
+    pbkdf2.pbkdf2(password, salt, PBKDF_ITERATIONS, KEY_SIZE, DIGEST, (err, key) => {
+      if (err) return reject(err)
+      resolve(key)
+    })
+  })
+}
+
+// hotfix alternative to crypto-random-string
+const cryptoRandomString = ({length}: {length: number}): string => {
+  const randBytes = Buffer.alloc(length)
+  window.crypto.getRandomValues(randBytes)
+  return randBytes.toString('hex')
+}
+
+export async function encryptWithPassword(
```

Comment:
seems worth covering with unit tests so we are sure that the implementation doesn't accidentally change which would otherwise be hard to spot

Reply:
added in #8edba1a1a0e31a288a163ff247218ab360d45a06